src/main/java/com/epam/ta/reportportal/core/integration/impl/CreateIntegrationHandlerImpl.java

```diff
@@ -100,4 +101,43 @@ public OperationCompletionRS createProjectIntegration(ReportPortalUser.ProjectDe
 		return new OperationCompletionRS("Integration with id = " + integration.getId() + " has been successfully created.");
 	}
 
+	@Override
+	public OperationCompletionRS updateGlobalIntegration(Long id, UpdateIntegrationRQ updateRequest) {
+
+		ReportPortalIntegrationEnum reportPortalIntegration = ReportPortalIntegrationEnum.findByName(updateRequest.getIntegrationName())
+				.orElseThrow(() -> new ReportPortalException(ErrorType.INTEGRATION_NOT_FOUND, updateRequest.getIntegrationName()));
+
+		Integration integration = integrationServiceMapping.get(reportPortalIntegration)
+				.updateGlobalIntegration(id, updateRequest.getIntegrationParams());
+
+		integration.setEnabled(updateRequest.getEnabled());
+
+		integrationRepository.save(integration);
+
+		return new OperationCompletionRS("Integration with id = " + integration.getId() + " has been successfully updated.");
+	}
+
+	@Override
+	public OperationCompletionRS updateProjectIntegration(Long id, ReportPortalUser.ProjectDetails projectDetails,
+			UpdateIntegrationRQ updateRequest, ReportPortalUser user) {
+
+		Project project = projectRepository.findById(projectDetails.getProjectId())
+				.orElseThrow(() -> new ReportPortalException(ErrorType.PROJECT_NOT_FOUND, projectDetails.getProjectId()));
+
+		ReportPortalIntegrationEnum reportPortalIntegration = ReportPortalIntegrationEnum.findByName(updateRequest.getIntegrationName())
+				.orElseThrow(() -> new ReportPortalException(ErrorType.INTEGRATION_NOT_FOUND, updateRequest.getIntegrationName()));
+
+		Integration integration = integrationServiceMapping.get(reportPortalIntegration)
+				.updateProjectIntegration(id, projectDetails, updateRequest.getIntegrationParams());
+
```

Comment:
in my opinion, if you are deviding code with idents it is better to group it by logic
here could be grouped as actions with integration
what do you think?

Reply:
done